python/lsst/ts/watcher/base/alarm.py

```diff
@@ -107,10 +111,69 @@ def acknowledge(self, severity, user):
         self._run_callback()
         return True
 
+    def mute(self, duration, severity, user):
+        """Mute this alarm for a specified duration.
+
+        Parameters
+        ----------
+        duration : `float`
+            How long to mute the alarm (sec)
+        severity : `lsst.ts.idl.enums.Watcher.AlarmSeverity` or `int`
+            Severity to mute. If the severity goes above
+            this level the alarm will be shown.
+        user : `str`
+            Name of user; used to set muted_by.
+
+        Raises
+        ------
+        ValueError
+            If ``duration <= 0``, ``severity == AlarmSeverity.NONE``
+            or ``severity`` is not a valid ``AlarmSeverity`` enum value.
+        """
+        if duration <= 0:
+            raise ValueError(f"duration={duration} must be positive")
+        severity = AlarmSeverity(severity)
+        if severity == AlarmSeverity.NONE:
+            raise ValueError(f"severity={severity!r} must be > NONE")
+        self.muted_by = user
+        self.muted_severity = severity
+        curr_tai = salobj.tai_from_utc(time.time())
+        self.timestamp_unmute = curr_tai + duration
+        self.unmute_task.cancel()
+        self.unmute_task = asyncio.create_task(self.unmute_after(duration=duration))
+        self._run_callback()
+
+    def unmute(self, run_callback=True):
+        """Unmute this alarm.
```

Comment:
Reading this right, you can mute alarms of a given severity, but you can only unmute "all" muted levels? Is it possible to have two "mute"ings active at a time? E.g. what happens if one user does `mute(500, WARNING, "me")` and another then does `mute(100, SERIOUS, "you")`?

Reply:
An alarm can only be in one muted state. The most recent call wins. In practice it is impossible for an operator to mute an already-muted alarm (unless its current value is higher than muted_severity, at which point it is displayed as a normal alarm), because the alarm has to be displayed in LOVE, the operator interface, and it won't be if it is muted.

Reply:
Also I don't mute an alarm "of a given severity" but "for a given severity". In other words the alarm will be muted as long as its current or max severity is less than or equal to its mute severity.

Reply:
I'll try to clarify this a bit in the doc string for mute. It already said something like this, but I think this is a minor improvement:
```
        severity : `lsst.ts.idl.enums.Watcher.AlarmSeverity` or `int`
            Severity to mute. If the alarm's current or max severity
            goes above this level the alarm should be displayed.
```

Reply:
I will also add this note to `mute`:
```
        Notes
        -----
        An alarm cannot have multiple mute levels and durations.
        If mute is called multiple times, the most recent call
        overwrites information from earlier calls.
```